release/long_running_distributed_tests/workloads/pytorch_pbt_failure.py

```diff
@@ -49,14 +81,86 @@
     },
 )
 
+_NO_RESTORE_FROM_FRESH_CKPT_ID_THRESHOLD = 5
+
+_RESTORE_FROM_FRESH_ERR_MSG = """
+All trials have at least reached checkpoint id 5. Not expecting any trial to
+go back to checkpoint id 0 again. This error could indicate an incorrect checkpoint
+and restore behavior.
+"""
+
+_CKPT_ID_10 = 10
+_TIME_TO_REACH_CKPT_ID_10_IN_S = 3600
+
+_NOT_REACH_CKPT_ID_10_ERR_MSG = """
+Expecting to see all trials to reach a minimal checkpoint id of 10 by 1 hour.
+"""
+
+
+class InspectCheckpointIdCallback(Callback):
+    """A callback for asserting the right checkpoint restore behavior.
+
+    Also outputs an artifact that dumps `self._checkpoints` to `/tmp/ckpt_ids.txt`.
+    """
+
```

Comment:
Should we still split this into two PRs?

Reply:
reverting changes in this file to separate the PRs.